google-cloud-pubsub/support/doctest_helper.rb

```diff
@@ -416,6 +416,20 @@ def mock_pubsub
     end
   end
 
+  doctest.skip "Google::Cloud::Pubsub::Topic#publish_async" do
```

Comment:
The implementation has changed so much between the previous PR and this one, that the previous doc test configuration was no longer applicable. @quartzmo and I will need to create a strategy for the AsyncPublisher doc tests, but we will do that in a later PR, along with the final documentation, after the behavior added in this PR is accepted.